### ISSUE TITLE
Tweet filter

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/anatine#73]
Would be nice to be able to hide tweets in the timeline that matches a string/regex. For example, filter out tweets about a topic you don't care about.

-

Prior art:

- http://thesweetsetup.com/quick-tip-use-tweetbot-mute-unwanted-tweets/
- http://tidbits.com/article/15903
- https://github.com/justin/SilencedBots/blob/master/RegEx-Filters.md
- https://chrome.google.com/webstore/detail/open-tweet-filter/bdmjagdcpkfpebaaffpafncgkleijako?hl=en
- https://github.com/rstuven/OpenTweetFilter

**Comment by SamVerschueren** [comment]
Would it be possible to hide adds as well?

**Comment by sindresorhus** [comment]
@SamVerschueren Sure, but are there any ads in https://mobile.twitter.com yet?

**Comment by SamVerschueren** [comment]
Yes, they are marked with something like `Featured by...` or something (not sure about the translation)

**Comment by paulmolluzzo** [comment]
'Promoted' is how they mark them.

I'd like to hide Polls entirely too, but we're getting into the opinionated changes that we saw in `refined-github` with the news feed. If we can build in a toggle for certain tweet types that might head off the request we might get.

- Promotes
- Polls
- "So and So liked this random tweet"

These aren't "content" specific, so maybe it's a class name we can find and hide.

**Comment by sindresorhus** [comment]
@SamVerschueren Can you open a new issue about it? It's something we can just do automagically. Also include the HTML of the ad in the issue.

**Comment by sindresorhus** [comment]
@paulmolluzzo I think ads can just be hidden no matter what. Polls need to be an option though. I don't mind them. Why do you want to hide them?

**Comment by paulmolluzzo** [comment]
@sindresorhus they just don't provide any value to me. Half of them are silly questions anyway, and the few that aren't I don't want to answer.
